src/instrumenter/transpile.ts

```diff
@@ -443,19 +491,128 @@ function transpileLet(expr: SLet, ctx: TranspilingContext): Expression {
     return ctx.refBinding(letVarName);
 }
 
+function makeForLoop(
+    ctx: TranspilingContext,
+    iterVarName: string,
+    iterVarType: TypeNode,
+    start: SNode,
+    end: SNode | Expression
+): ForStatement {
+    const factory = ctx.factory;
+
+    ctx.addBinding(iterVarName, transpileType(iterVarType, ctx.factory));
+    // Iteration variable initialization statmenet
+    const initStmt = factory.makeExpressionStatement(
+        factory.makeAssignment("<missing>", "=", ctx.refBinding(iterVarName), transpile(start, ctx))
+    );
+
+    // Loop condition
+    const iterCond = factory.makeBinaryOperation(
+        "<missing>",
+        "<",
+        ctx.refBinding(iterVarName),
+        end instanceof Expression ? end : transpile(end, ctx)
+    );
+
+    // Iteration variable increment
+    const incStmt = factory.makeExpressionStatement(
+        factory.makeUnaryOperation("<missing>", false, "++", ctx.refBinding(iterVarName))
+    );
+
+    const body: Block = factory.makeBlock([]);
```

Comment:
Type should be infered here, so no need for `Block`.

Reply:
Fixed in a97c7d9.